src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs

```diff
@@ -274,30 +280,42 @@ private static IEnumerable<IExternalSearchQueryResult> InternalExecuteSearch(IEx
                     placeIdRequest.AddParameter("locationbias", $"point:{splitCoordinates[0]}, {splitCoordinates[1]}");
                 }
 
-                //var locationName = string.Join("", query.QueryParameters["locationName"]);
-                //var transformedCoordinates = string.Join("", query.QueryParameters["coordinates"]);
-                //var splitCoordinates = transformedCoordinates.Split(',');
-                //placeIdRequest.AddParameter("input", locationName);
-                //placeIdRequest.AddParameter("locationbias", $"point:{splitCoordinates[0]}, {splitCoordinates[1]}");
                 placeIdRequest.AddParameter("inputtype", "textquery");
             }
 
-            var placeIdResponse = client.ExecuteTaskAsync<PlaceIdResponse>(placeIdRequest).Result;
-            if (placeIdResponse.Data.Status.Equals("REQUEST_DENIED"))
+            IRestResponse<PlaceIdResponse> placeIdResponse = null;
+
+            try
+            {
+                context.Log.LogInformation("Making Google Maps call. Request: ", JsonUtility.Serialize(placeIdRequest.Parameters));
+                placeIdResponse = client.ExecuteTaskAsync<PlaceIdResponse>(placeIdRequest).Result;
+            }
+            catch(Exception exception)
+            {
+                context.Log.LogError("Could not return PlaceIdResponse from Google Maps", exception);
+            }
+
+            if (placeIdResponse == null)
             {
                 yield break;
             }
 
+            if (placeIdResponse.Data.status.Equals("REQUEST_DENIED"))
+            {
+                context.Log.LogError("REQUEST DENIED by Google Maps");
+                yield break;
+            }
+
             if (placeIdResponse.StatusCode == HttpStatusCode.OK)
             {
                 if (placeIdResponse.Data != null && isCompany == false)
                 {
                     var request = new RestRequest(placeDetailsEndpoint, Method.GET);
-                    foreach (var placeId in placeIdResponse.Data.Candidates)
+                    foreach (var placeId in placeIdResponse.Data.results)
```

Comment:
313 to 318 is duplicated with 341 to 446, this is bad practice, should be extracted to a function or outside of the foreach.